operators/src/source/gdal_source/loading_info.rs

```diff
@@ -0,0 +1,535 @@
+            info: GdalLoadingInfoPartIterator::Static { parts },
+        })
+    }
+
+    async fn result_descriptor(&self) -> Result<RasterResultDescriptor> {
+        Ok(self.result_descriptor.clone())
+    }
+
+    fn box_clone(
+        &self,
+    ) -> Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>> {
+        Box::new(self.clone())
+    }
+}
+
+/// Meta data for a regular time series that begins (is anchored) at `start` with multiple gdal data
+/// sets `step` time apart. The `time_placeholders` in the file path of the dataset are replaced with the
+/// specified time `reference` in specified time `format`.
+// TODO: `start` is actually more a reference time, because the time series also goes in
+//        negative direction. Maybe it would be better to have a real start and end time, then
+//        everything before start and after end is just one big nodata raster instead of many
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetaDataRegular {
+    pub result_descriptor: RasterResultDescriptor,
+    pub params: GdalDatasetParameters,
+    pub time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    pub start: TimeInstance,
+    pub step: TimeStep,
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetaDataRegular
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let snapped_start = self
+            .step
+            .snap_relative(self.start, query.time_interval.start())?;
+
+        let snapped_interval =
+            TimeInterval::new_unchecked(snapped_start, query.time_interval.end()); // TODO: snap end?
+
+        let time_iterator =
+            TimeStepIter::new_with_interval_incl_start(snapped_interval, self.step)?;
+
+        Ok(GdalLoadingInfo {
+            info: GdalLoadingInfoPartIterator::Dynamic(DynamicGdalLoadingInfoPartIterator::new(
+                time_iterator,
+                self.params.clone(),
+                self.time_placeholders.clone(),
+                self.step,
+                query.time_interval.end(),
+            )?),
+        })
+    }
+
+    async fn result_descriptor(&self) -> Result<RasterResultDescriptor> {
+        Ok(self.result_descriptor.clone())
+    }
+
+    fn box_clone(
+        &self,
+    ) -> Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>> {
+        Box::new(self.clone())
+    }
+}
+
+/// Meta data for 4D `NetCDF` CF datasets
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetadataNetCdfCf {
+    pub result_descriptor: RasterResultDescriptor,
+    pub params: GdalDatasetParameters,
+    pub start: TimeInstance,
+    pub end: TimeInstance, // TODO: Use this or time dimension size (number of steps)?
+    pub step: TimeStep,
+    pub band_offset: usize,
```

Comment:
what is this? Add a comment how it works

Reply:
5692f8d